### PR TITLE
Modification to support Mariner distribution

### PR DESCRIPTION
**Pull request by johnsongeorge-w**

### CODE REVIEW

File: Testscripts/Linux/KDUMP-Config.sh [hunk 1]
```diff
@@ -348,12 +408,18 @@ Config_"${OS_FAMILY}"
 # Remove old crashkernel params
 sed -i --follow-symlinks "s/crashkernel=\S*//g" $boot_filepath
 
+# Remove console params; It could interfere with the testing
+sed -i --follow-symlinks "s/console=\S*//g" $boot_filepath
```
Comment (LiliDeng) [inline Testscripts/Linux/KDUMP-Config.sh]: no need this now

Comment (johnsongeorge-w) [reply Testscripts/Linux/KDUMP-Config.sh]: Fixed

File: Libraries/TestHelpers.psm1 [hunk 2]
```diff
@@ -315,6 +315,11 @@ Function Copy-RemoteFiles($uploadTo, $downloadFrom, $downloadTo, $port, $files,
 	}
 	elseif ($download) {
 		foreach ($file in $fileList) {
+			if ($username -ne "root") {
+				$chown_cmd = "chown -Rf ${username}: $file"
+				Write-LogInfo "$chown_cmd"
+				$out = Run-LinuxCmd -username $username -password $password -ip $downloadFrom -port $port -command "$chown_cmd" -runAsSudo
+			}
 			Download-RemoteFile -downloadFrom $downloadFrom -downloadTo $downloadTo -port $port -file $file -username $username `
```
Comment (LiliDeng) [inline Libraries/TestHelpers.psm1]: these lines will break some platform, but we need figure out a way to fix the issue when umask value is not 022 by default.

**Comment by johnsongeorge-w** [comment]
Test P0 & P1 testcases and result looks good.

**Review by LiliDeng — DISMISSED** [review]

File: Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh [hunk 3]
```diff
@@ -84,37 +84,40 @@ for fs in "${fileSystems[@]}"; do
...
+    check_exit_status "Mount partition" "failed"
 
     # Read/Write mount point
     ./STOR_VHDXResize_ReadWrite.sh
 
     umount $mntDir
-    check_exit_status "Unmount partition" "exit"
-
+    check_exit_status "Unmount partition" "failed"
+	sleep 30
```
Comment (LiliDeng) [inline Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh]: indent

File: Testscripts/Linux/nginx-community-test.sh [hunk 4]
```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (LiliDeng) [inline Testscripts/Linux/nginx-community-test.sh]: which case use this nginx-community-test.sh ?

File: Testscripts/Windows/SELINUX-REGRESSION-TESTSUITE.ps1 [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
```
Comment (LiliDeng) [inline Testscripts/Windows/SELINUX-REGRESSION-TESTSUITE.ps1]: not add the  case into xml file?

File: Testscripts/Linux/timesync_chrony.sh [hunk 6]
```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
```
Comment (LiliDeng) [inline Testscripts/Linux/timesync_chrony.sh]: same for this file

File: Testscripts/Linux/xfstesting.sh [hunk 7]
```diff
@@ -70,6 +78,12 @@ ConfigureXFSTestTools() {
         install_nvme_cli
     fi
     modprobe btrfs
+    wget https://phoenixnap.dl.sourceforge.net/project/libuuid/libuuid-1.0.3.tar.gz
+
+    tar -xvf libuuid-1.0.3.tar.gz
+    cd libuuid-1.0.3
+    ./configure && make && make install
+    cd ..
```
Comment (LiliDeng) [inline Testscripts/Linux/xfstesting.sh]: above steps may only for mariner.

File: Testscripts/Linux/xfstesting.sh [hunk 8]
```diff
@@ -237,15 +251,16 @@ Main() {
             # Azure provides links in /dev/disk/azure/ to /dev/sd[a-z] devices
             # this is needed as some 5+ kernels do not assign drive letters in any particular order
             # (previously the test disk was /dev/sdc)
-            testDisk=$(readlink -f /dev/disk/azure/scsi1/* | sed "s@/dev/@@g" | head -1)
+            #testDisk=$(readlink -f /dev/disk/azure/scsi1/* | sed "s@/dev/@@g" | head -1)
+            testDisk="sdc"
```
Comment (LiliDeng) [inline Testscripts/Linux/xfstesting.sh]: we can't hard code the disk name, it may change.

File: Testscripts/Linux/xfstesting.sh [hunk 9]
```diff
@@ -237,15 +251,16 @@ Main() {
...
             ConfigureDisks "${testDisk}" "${testDisk}1" "${testDisk}2" "$FSTYP" "test" "scratch"
             # Configure disk in env and xfstests config file
             TEST_DEV="/dev/${testDisk}1"
             echo "TEST_DEV=${TEST_DEV}" >> ${XFSTestConfigFile}
         fi
     fi
     # Copy config file into the xfstests folder
-    dos2unix ${XFSTestConfigFile}
+    sed -i 's/\r$//' ${XFSTestConfigFile}
```
Comment (LiliDeng) [inline Testscripts/Linux/xfstesting.sh]: have we checked that dos2unix still absent in mariner?

File: Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh (hunk 3 above)
Comment (johnsongeorge-w) [reply Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh]: Fixed

File: Testscripts/Linux/nginx-community-test.sh (hunk 4 above)
Comment (johnsongeorge-w) [reply Testscripts/Linux/nginx-community-test.sh]: Dropped

File: Testscripts/Windows/SELINUX-REGRESSION-TESTSUITE.ps1 (hunk 5 above)
Comment (johnsongeorge-w) [reply Testscripts/Windows/SELINUX-REGRESSION-TESTSUITE.ps1]: Dropped

File: Testscripts/Linux/timesync_chrony.sh (hunk 6 above)
Comment (johnsongeorge-w) [reply Testscripts/Linux/timesync_chrony.sh]: Dropped

File: Testscripts/Linux/xfstesting.sh (hunk 7 above)
Comment (johnsongeorge-w) [reply Testscripts/Linux/xfstesting.sh]: reverted to previous commit.

File: Testscripts/Linux/xfstesting.sh (hunk 8 above)
Comment (johnsongeorge-w) [reply Testscripts/Linux/xfstesting.sh]: Reverted to previous commit.

File: Testscripts/Linux/xfstesting.sh (hunk 9 above)
Comment (johnsongeorge-w) [reply Testscripts/Linux/xfstesting.sh]: Reverted to previous commit.

**Review by LiliDeng — DISMISSED** [review]

File: TestProviders/AzureProvider.psm1 [hunk 10]
```diff
@@ -61,6 +61,10 @@ Class AzureProvider : TestProvider {
 					$deployedGroups = $isAllDeployed[1]
 					$DeploymentElapsedTime = $isAllDeployed[3]
 					$allVMData = Get-AllDeploymentData -ResourceGroups $deployedGroups
+
+					# After successful deployment, configure the umask for Linux distro
+					$null = Configure-UmaskInVMs $allVMData $global:user $global:password
```
Comment (LiliDeng) [inline TestProviders/AzureProvider.psm1:66]: I didn't see you pass sshport, though it is global variable.

Comment (LiliDeng) [reply TestProviders/AzureProvider.psm1:66]: please double check the pass in and defined parameters.

File: Testscripts/Linux/enable_passwordless_root.sh [hunk 11]
```diff
@@ -52,4 +52,11 @@ else
...
     fi
     echo "KEY_GENERATED_SUCCESSFULLY"
-fi
\ No newline at end of file
+fi
+
+if [[ -d /run/systemd/system ]];then
+   systemctl stop iptables 2> /dev/null && systemctl disable iptables
+else
+    service iptables stop 2> /dev/null && service iptables disable
```
Comment (LiliDeng) [inline Testscripts/Linux/enable_passwordless_root.sh:60]: indent?

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**